tools/run_tests/jobset.py

```diff
@@ -254,14 +254,21 @@ def stdout(self=self):
       stdout = self._tempfile.read()
       self.result.message = stdout[-_MAX_RESULT_SIZE:]
       return stdout
+    is_http2 = False
+    msg = ''
+    if 'http2' in self._spec.shortname:
```

Comment:
So I looked into this and why it may be happening.  It looks like change db21899edcd663bf941b0c7ff07d5a3218908c9b (which delays reading stdout) can potentially result in never reading stdout?  I think the correct fix is to just forcibly read the whole output at the end.

Looking more thoroughly at this change it appears that it won't be sufficient.  Any other test that runs very quickly will likely have the same problem.  For that matter, the code that is even reading stdout is horribly inefficient; it loads the entire file into memory, and then makes a copy of the slice its interested in?  What!?

This whole testing framework could use some love.   I agree with @ctiller on this one, a proper fix is needed.